pkg/metrics/metrics_test.go

```diff
@@ -0,0 +1,64 @@
+package metrics
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	dto "github.com/prometheus/client_model/go"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+func TestMetrics(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Metrics Suite")
+}
+
+func findMetric(src []*dto.MetricFamily, query string) *dto.MetricFamily {
+	for _, s := range src {
+		if s.Name != nil && *s.Name == query {
+			return s
+		}
+	}
+	return nil
+}
+
+var _ = Describe("Metrics", func() {
+	value := 12
+	sr := "simple-kmod"
+	state := "templates/0000-buildconfig.yaml"
+	kind := "BuildConfig"
+	name := "simple-kmod-driver-build"
+	namespace := "openshift-special-resource-operator"
+
+	m := New()
+	m.SetSpecialResourcesCreated(value)
```

Comment:
Could we use a different value in each call to make sure that the methods target the correct metric?

Reply:
Yes, of course - changed